src/content/GraphError.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @module GraphError
+ */
+
+import { Parsable, ParseNode, AdditionalDataHolder, BackedModel, ApiError } from "@microsoft/kiota-abstractions";
+
+/**
+ * @interface
+ * Signature represents the structure of an error response
+ */
+export interface GraphError extends ApiError, AdditionalDataHolder, BackedModel {
```

Comment:
can you provide some more context on why we need to re-define that type here please?

Reply:
This is to provide a default implementation of errors when none is provided by the user. The ApiError class does not have additional data. I could remove this make the error constructor mandatory

Reply:
I think there are two cases we need to think about:
- the user is NOT using the service lib, in which case, should we let them using the page iterator? I'd argue that no since a lot of the infrastructure here assumes we'll deserialize to a generated model, instead of raw JSON.
- the user IS using the service lib, in which case can we make their lives easier? yes! we can make the parameter mandatory here, remove the duplicated definition, and have a sub-class in the service libs which provides that parameter/error mapping for the user.

Thoughts?

Reply:
I do agree on both options.

1. For users not using the service lib they will need to offer an implementation of there own factory. 
2. Users can by default be guided to use the "OdataError" factory